wordpress-mobile/WordPress-FluxC-Android
# MSD - Stats Card - Handle Error Inside Cards Response

Parent https://github.com/wordpress-mobile/WordPress-Android/issues/15749

This draft PR:
- Updates the `today's stats` and `posts` cards responses to include error.
- Updates the `today's stats` and `posts` card model to include error.
- Maps error in responses to an error in models.
- Makes no changes to storing and retrieving logic for dashboard cards into the database. This means the error inside the card model will get saved into the database whenever it is present and returned to the client as part of the card model. We can discuss any changes to this behavior.

------------------------------------------------------------------------------------------

To test:

- There is actually nothing to test in this PR, just review the updated `CardsRestClient` and `CardsStore` classes and make sure that the accompanying tests are working as expected.

------------------------------------------------------------------------------------------

Notes: 

- As the differential is still to be deployed, the `fetchCards` endpoint is not replaced with `v1_1` version yet to avoid rest route not found errors.
- To manual test: Apply D73653-code on your sandbox, proxy your device on it. Replace the `fetchCards` endpoint with `v1_1` version.
- Let's keep the PR in draft mode till the corresponding `WordPress-Android` PR is up and we've tested cards with errors feature.
